instrumentation/github.com/emicklei/go-restful/otelrestful/internal/semconv/httpconv.go

```diff
@@ -558,11 +558,17 @@ func (n CurrentHTTPClient) TraceAttributes(host string) []attribute.KeyValue {
 	}
 }
 
-func (n CurrentHTTPClient) scheme(https bool) attribute.KeyValue { // nolint:revive
-	if https {
-		return semconvNew.URLScheme("https")
+func (n CurrentHTTPClient) scheme(req *http.Request) attribute.KeyValue {
+	if req.URL == nil {
```

Comment:
This way of doing the condition looks weird. How about:

```golang
if req.URL != nil && req.URL.Scheme != "" {
	return semconvNew.URLScheme(req.URL.Scheme)
}
if req.TLS != nil {
	return semconvNew.URLScheme("https")
}
return semconvNew.URLScheme("http")
```

Reply:
done